CourseManager/ViewModels/AddOrEditCourseViewModel.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  BookViewModel.swift
+//  CourseManager
+//
+//  Created by Summer Crow on 2022-09-30.
+//
+
+import Combine
+import Firebase
+import Foundation
+
+class AddOrEditCourseViewModel: ObservableObject {
+    @Published var course: CourseInformation = .init(id: "", name: "", url: "", gitHubURL: "", status: .todo, dateOfCompletion: Date(), comments: "")
+    private var db = Firestore.firestore()
+    private var cancellables = Set<AnyCancellable>()
+
+    func addCourse(course: CourseInformation) {
+        do {
+            try db.collection("courses").addDocument(from: course)
+        } catch {
+            print(error)
+        }
+    }
+
+    func edit() {
+        let idRef = db.collection("courses").document(course.id!)
```

Comment:
Force cast isn't what you should use in this situation. There is no proof, that it isn't nil. Even if it is and you will decide to change it, then it will crash app

Reply:
Ah yes, updated to : 
guard let courseID = course.id else {
            return
        }
let idRef = CourseRepository.collection.document(courseID)